features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -594,11 +594,13 @@ nsapi_error_t mbed_lwip_emac_init(emac_interface_t *emac)
 // Backwards compatibility with people using DEVICE_EMAC
 nsapi_error_t mbed_lwip_init(emac_interface_t *emac)
 {
-    nsapi_error_t ret;
+    nsapi_error_t ret = NSAPI_ERROR_OK;
     mbed_lwip_core_init();
-    ret = mbed_lwip_emac_init(emac);
-    if (ret == NSAPI_ERROR_OK) {
-        netif_inited = true;
+    if(netif_inited == false){
```

Comment:
Formatting - missing spaces around `()`.

I'd kind of like a sanity check on the other case where we are already inited, rather than always just returning OK. Something like

    if (netif_is_ppp || emac != lwip_netif.state) {
        ret = NSAPI_ERROR_PARAMETER;
    }

to catch someone trying to register a different EMAC, which we couldn't cope with.

Mind you, thinking about it, if you're making this change because you're calling this on every `connect` to your driver, you shouldn't be doing that - this should be a one-off operation - `if (!inited)` in your driver's connect, or do it on construction. That will match better the coming multiple-interface world (#5558).

So I'm wondering if we should actually be faulting any second init call with NSAPI_ERROR_PARAMETER - effectively saying "we don't (yet) support a second interface".

Reply:
I tried modifying my driver to call the mbed_lwip_init function only once but it still fails connectivity test for bring the network up and down twice. I tried putting the call in constructor, I also made the check as you mentioned but the test still fails.

Reply:
I'm not familiar with this test - is there a possibility that the mbed_lwip_xxx stuff is fundamentally bugged in some way that prevents this passing? Do any other EMAC drivers pass this test?

@SeppoTakalo - do we have an equivalent test?

Reply:
Can't see any obvious reason for failure - the bringup/bringdown logic in lwip_stack.c for a 2nd go around is much the same between native lwIP drivers and EMAC drivers, and I know native lwIP drivers are passing.

Calling mbed_lwip_bringdown (in your disconnect) followed by mbed_lwip_bringup (in your connect) should work.

Reply:
Ok I called the bringdown in my disconnect, it worked, shall I update the changes in this PR or create a new one?